### PR TITLE
Replace DeveloperError with Check in Matrix4.js

### PR DESCRIPTION
**Pull request by shehzan10**
Part of #4794

### CODE REVIEW

File: Source/Core/Matrix4.js [hunk 1]
```diff
@@ -104,13 +106,9 @@ define([
      */
     Matrix4.pack = function(value, array, startingIndex) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(value)) {
-            throw new DeveloperError('value is required');
-        }
+        Check.typeOf.object(value, 'value');
```
Comment (lilleyse) [inline Source/Core/Matrix4.js]: Remove whitespace.

File: Source/Core/Matrix4.js [hunk 2]
```diff
@@ -675,9 +643,7 @@ define([
         if (far <= 0.0) {
```
Comment (lilleyse) [inline Source/Core/Matrix4.js]: Can these be checked with `Check.numeric.minimum` and `maximum`? I'm not sure what the best way to handle `<=` is though, something like `Check.numeric.minimum(CesiumMath.EPSILON7)` would be clunky.

File: Source/Core/Matrix4.js [hunk 3]
```diff
@@ -1171,18 +1081,14 @@ define([
...
-        }
-        if (!defined(result)) {
-            throw new DeveloperError('result is required');
-        }
+        Check.typeOf.object(matrix, 'matrix');
+        Check.typeOf.object(cartesian, 'cartesian');
+
+        Check.typeOf.number(index, 'index');
+        Check.numeric.minimum(index, 0);
+        Check.numeric.maximum(index, 3);
```
Comment (lilleyse) [inline Source/Core/Matrix4.js]: Put `index` check before `cartesian`.

File: Source/Core/Matrix4.js [hunk 4]
```diff
@@ -1320,18 +1217,14 @@ define([
...
+
+        Check.typeOf.object(result, 'result');
```
Comment (lilleyse) [inline Source/Core/Matrix4.js]: Same comment.

File: Source/Core/Matrix4.js (hunk 2 above)
Comment (shehzan10) [inline Source/Core/Matrix4.js]: Right, thats why I left it as is until we can figure out a more robust solution.

File: Source/Core/Matrix4.js (hunk 4 above)
Comment (shehzan10) [inline Source/Core/Matrix4.js]: I'm assuming this is misplaced and was for lines 1089/1225.

Comment (lilleyse) [inline Source/Core/Matrix4.js]: Ah yes

**Comment by lilleyse** [comment]
Nice work @shehzan10. Even though you're added to the contributors list in https://github.com/AnalyticalGraphicsInc/cesium/pull/4831, may as well merge this now.

**Pull request merged**